tests/Aspire.Hosting.Redis.Tests/RedisFunctionalTests.cs

```diff
@@ -0,0 +1,306 @@
+            });
+
+            hb.AddRedisClient(redis2.Resource.Name);
+
+            using (var host = hb.Build())
+            {
+                await host.StartAsync();
+
+                var redisClient = host.Services.GetRequiredService<IConnectionMultiplexer>();
+
+                var db = redisClient.GetDatabase();
+
+                var value = await db.StringGetAsync("key");
+
+                Assert.Equal("value", value);
+            }
+
+            await app.StopAsync();
+        }
+
+        try
+        {
+            File.Delete(bindMountPath);
+        }
+        catch
+        {
+            // Don't fail test if we can't clean the temporary folder
+        }
+    }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task PersistenceIsDisabledByDefault()
+    {
+        // Checks that without enabling Redis Persistence the tests fail
+
+        var builder1 = TestDistributedApplicationBuilder.Create();
+        var redis1 = builder1.AddRedis("redis");
+
+        using (var app = builder1.Build())
+        {
+            await app.StartAsync();
+
+            var hb = Host.CreateApplicationBuilder();
+
+            // BGSAVE is only available in admin mode, enable it for this instance
+            hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+            {
+                [$"ConnectionStrings:{redis1.Resource.Name}"] = $"{await redis1.Resource.GetConnectionStringAsync()},allowAdmin=true"
+            });
+
+            hb.AddRedisClient(redis1.Resource.Name);
+
+            using (var host = hb.Build())
+            {
+                await host.StartAsync();
+
+                var redisClient = host.Services.GetRequiredService<IConnectionMultiplexer>();
+
+                var db = redisClient.GetDatabase();
+
+                await db.StringSetAsync("key", "value");
+            }
+
+            await app.StopAsync();
+        }
+
+        var builder2 = TestDistributedApplicationBuilder.Create();
+        var redis2 = builder2.AddRedis("redis");
+
+        using (var app = builder2.Build())
+        {
+            await app.StartAsync();
+
+            var hb = Host.CreateApplicationBuilder();
+
+            hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+            {
+                [$"ConnectionStrings:{redis2.Resource.Name}"] = await redis2.Resource.GetConnectionStringAsync()
+            });
+
+            hb.AddRedisClient(redis2.Resource.Name);
+
+            using (var host = hb.Build())
+            {
+                await host.StartAsync();
+
+                var redisClient = host.Services.GetRequiredService<IConnectionMultiplexer>();
+
+                var db = redisClient.GetDatabase();
+
+                var value = await db.StringGetAsync("key");
+
+                Assert.True(value.IsNull);
+            }
+
+            await app.StopAsync();
+        }
+    }
+
+    private static void AttemptDeleteDockerVolume(string volumeName)
```

Comment:
We will probably need this in other test projects. @eerhardt Any preference where to move it? Copy it for now?

Reply:
Probably can move it into https://github.com/dotnet/aspire/tree/main/tests/Aspire.Hosting.Tests/Utils. We already are using other Utils in there, so I assume all the other separate test assemblies will use it.